### PR TITLE
Allow running performance tests on Windows

### PR DESCRIPTION
**Pull request by wolfs**
(Supersedes #11157)

A step towards https://github.com/gradle/gradle-private/issues/1860.

Current problems with the performance tests:
- [x] For Android build, the AAPT2 Daemon startup fails, probably because of too long paths. See for example [this build](https://builds.gradle.org/buildConfiguration/Gradle_Check_IndividualPerformanceScenarioWorkersWindows/37375610?buildTab=overview).
  - Fixed by using `C:\gradle-perf` as checkout dir.
- [x] Make sure performance results from Windows and Linux performance tests are not mixed up in the performance database. That means that the history of a test running on Linux shouldn't show results from Windows and vice-versa.
  - Use `commits-windows`/`adhoc-windows`, etc. on Windows.
- [x] Kill other Gradle processes on Windows, see [here](https://github.com/gradle/gradle/blob/34537c967d5cc1ccbd6628b7a448ceeee998cdce/.teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt#L24-L35)
- [x] System property containing spaces is incorrectly escaped when passed to Gradle profiler. See for example [this build](https://builds.gradle.org/buildConfiguration/Gradle_Check_IndividualPerformanceScenarioWorkersWindows/37375786?buildTab=overview).
  - Spaces are removed.
- [x] Perf Android large doesn't work on Windows, since it expects `zip` to be available. See [here](https://github.com/gradle/perf-android-large/blob/android-36/build.gradle#L159-L163)
  - `7z` was used on Windows to perform the same action.
- [x] The Swift performance tests don't run on Windows and should be ignored there.
  - Done.
- [x] Some dependency management tests can't load dependencies from the embedded repository, check e.g. [this build](https://builds.gradle.org/buildConfiguration/Gradle_Check_IndividualPerformanceScenarioWorkersWindows/37375664?buildTab=overview).
  - Confirmed to be path length issue, fixed by shorter checkout dir.
- [x] Native tests fail with errors when linking, probably related to too long paths, see e.g. [this build](https://builds.gradle.org/buildConfiguration/Gradle_Check_IndividualPerformanceScenarioWorkersWindows/37375646?buildTab=overview).
  - Native tests are ignored.
- [x] Some other native sources don't compile on Windows, see [this build](https://builds.gradle.org/buildConfiguration/Gradle_Check_IndividualPerformanceScenarioWorkersWindows/37375800?buildTab=overview).
  - Native tests are ignored.

### CODE REVIEW

**Comment by wolfs** [comment]
Apart from the problems mentioned in the issue description, the performance tests seem to run mostly fine on Windows, see e.g. https://builds.gradle.org/buildConfiguration/Gradle_Util_Performance_PerformanceTestCoordinatorWindows/37375332?buildTab=overview for a run.

Some tests take very long, like `clean assemble with * build cache on largeJavaMultiproject`.

The most important problem to fix is the supposedly long path issues for the santa tracker builds. I don't think making `perf-android-large` work on Windows is necessarily a good investment of our time, though it also doesn't seem to be very hard.

Being able to build a baseline for PRs also seems like a good feature, though we can do without for now.

As next step I would suggest to select a subset of the tests which we want to run on Windows and run those at some intervals.

File: .teamcity/common/performance-test-extensions.kt [hunk 1]
```diff
@@ -39,14 +39,17 @@ fun BuildType.applyPerformanceTestSettings(os: Os = Os.linux, timeout: Int = 30)
...
+    "$task --baselines $baselines $extraParameters",
+    "-x prepareSamples"
+) + listOf(
+    "-PtestJavaHome" to testJavaHome,
+    "-Porg.gradle.performance.branchName" to "%teamcity.build.branch%",
+    "-Porg.gradle.performance.db.url" to "%performance.db.url%",
+    "-Porg.gradle.performance.db.username" to "%performance.db.username%",
+    "-Porg.gradle.performance.db.password" to "%performance.db.password.tcagent%",
+    "-PteamCityToken" to "%teamcity.user.bot-gradle.token%"
+    ).map { (key, value) -> """"$key=$value"""" }
```
Comment (wolfs) [inline .teamcity/common/performance-test-extensions.kt:52]: We should check here if `os.escapeKeyValuePair` works as well, and if it does use it.

**Comment by wolfs** [comment]
Some other tests are also much slower: `up-to-date assemble on largeJavaMultiproject (parallel = false)` runs in 4s on Linux and takes 16s on Windows per iteration.

**Comment by wolfs** [comment]
I think another thing which probably needs to be done is to track the operating system in the performance database, so Windows and Linux performance results are not mixed up.

**Comment by blindpirate** [comment]
Some progress:

- Regarding long path name, I use custom checkout dir `C:\gradle` and it works for at least one scenario: https://builds.gradle.org/viewLog.html?buildId=37492610&buildTypeId=Gradle_Check_IndividualPerformanceScenarioWorkersWindows&tab=buildResultsDiv , will run more scenarios.
- The special system property containing spaces was adjusted.
- Windows and Linux builds are distinguished via performance test channels.
- Native scenaios are ignored for now.

Still in progress:

- Try to use cygwin commands to do the same thing as `zip -d`.
- Kill all gradle processes on Windows.

**Comment by blindpirate** [comment]
@bot-gradle test RFN

**Comment by blindpirate** [comment]
@bot-gradle test RFN

**Comment by blindpirate** [comment]
Now all windows performance tests are able to run: https://builds.gradle.org/viewLog.html?buildId=37555094&buildTypeId=Gradle_Util_Performance_PerformanceTestCoordinatorWindows&tab=buildResultsDiv

File: .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt [hunk 2]
```diff
@@ -21,14 +21,23 @@ import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.script
 import model.CIBuildModel
 import model.StageNames
 
-val killAllGradleProcesses = """
+val killAllGradleProcessesLinux = """
```
Comment (lptr) [inline .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt:24]: Is this for Linux or UNIX/POSIX (meaning it should work on macOS too)?

Comment (lptr) [reply .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt:24]: I see we use `UnixLike` vs `Windows` elsewhere.

File: .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt [hunk 3]
```diff
@@ -20,24 +20,35 @@ class IndividualPerformanceScenarioWorkers(model: CIBuildModel, os: Os = Os.linu
...
         param("checks", "all")
         param("runs", "defaults")
         param("warmups", "defaults")
         param("scenario", "")
 
-        param("env.ANDROID_HOME", "/opt/android/sdk")
-        param("env.PATH", "%env.PATH%:/opt/swift/4.2.3/usr/bin")
+        param("env.ANDROID_HOME", os.androidHome)
+        if (os == Os.linux) {
+            param("env.PATH", "%env.PATH%:/opt/swift/4.2.3/usr/bin")
```
Comment (lptr) [inline .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt:41]: What about macOS?

Comment (lptr) [reply .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt:41]: I think this `if` should be a `switch` instead.

File: .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt [hunk 4]
```diff
@@ -20,24 +20,35 @@ class IndividualPerformanceScenarioWorkers(model: CIBuildModel, os: Os = Os.linu
...
+            param("env.PATH", "%env.PATH%;C:/Program Files/7-zip")
+        }
     }
 
     steps {
         script {
             name = "KILL_GRADLE_PROCESSES"
             executionMode = BuildStep.ExecutionMode.ALWAYS
-            scriptContent = killAllGradleProcesses
+            scriptContent = if (os == Os.windows) killAllGradleProcessesWindows else killAllGradleProcessesLinux
```
Comment (lptr) [inline .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt:51]: Can this be a property of the `common.Os` type instead? So here we'd say:

```suggestion
            scriptContent = os.killAllGradleProcesses
```

File: .teamcity/Gradle_Util_Performance/buildTypes/AdHocPerformanceScenarioWindows.kt [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                    """--scenarios "%scenario%" --warmups %warmups% --runs %runs% --checks %checks% --channel %channel% %flamegraphs% %additional.gradle.parameters%""",
+                    individualPerformanceTestJavaHome(Os.windows)
+                ) +
+                    buildToolGradleParameters(isContinue = false) +
+                    builtInRemoteBuildCacheNode.gradleParameters(Os.windows)
+                ).joinToString(separator = " ")
+        }
+        checkCleanM2()
+    }
+})
```
Comment (lptr) [inline .teamcity/Gradle_Util_Performance/buildTypes/AdHocPerformanceScenarioWindows.kt:61]: There's a lot of redundant code here with `AdHocPerformanceScenarioLinux.kt`, let's extract that out to a common superclass or something.

File: .teamcity/common/Os.kt [hunk 6]
```diff
@@ -16,6 +16,10 @@
 
 package common
 
-enum class Os(val agentRequirement: String, val ignoredSubprojects: List<String> = emptyList()) {
-    linux("Linux"), windows("Windows"), macos("Mac", listOf("integ-test", "native", "plugins", "resources", "scala", "workers", "wrapper", "platform-play", "tooling-native"))
+enum class Os(val agentRequirement: String, val ignoredSubprojects: List<String> = emptyList(), val androidHome: String = "/opt/android/sdk") {
```
Comment (lptr) [inline .teamcity/common/Os.kt:19]: I think it would be better to write `/opt/android/sdk` out instead of relying on this default.

File: .teamcity/common/performance-test-extensions.kt (hunk 1 above)
Comment (lptr) [reply .teamcity/common/performance-test-extensions.kt:52]: I think we should follow up on @wolfs' comment here.

File: .teamcityTest/Gradle_Check_Tests/PerformanceTestBuildTypeTest.kt [hunk 7]
```diff
@@ -61,12 +61,12 @@ class PerformanceTestBuildTypeTest {
...
-                "-Porg.gradle.performance.db.username=%performance.db.username%",
-                "-Porg.gradle.performance.db.password=%performance.db.password.tcagent%",
-                "-PteamCityToken=%teamcity.user.bot-gradle.token%",
-                "-PtestJavaHome=%linux.java8.oracle.64bit%",
+                """"-PtestJavaHome=%linux.java8.oracle.64bit%"""",
+                """"-Porg.gradle.performance.branchName=%teamcity.build.branch%"""",
+                """"-Porg.gradle.performance.db.url=%performance.db.url%"""",
+                """"-Porg.gradle.performance.db.username=%performance.db.username%"""",
+                """"-Porg.gradle.performance.db.password=%performance.db.password.tcagent%"""",
+                """"-PteamCityToken=%teamcity.user.bot-gradle.token%"""",
```
Comment (lptr) [inline .teamcityTest/Gradle_Check_Tests/PerformanceTestBuildTypeTest.kt:69]: Why is this change necessary?

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt [hunk 8]
```diff
@@ -269,34 +270,36 @@ class PerformanceTestPlugin : Plugin<Project> {
             createDistributedPerformanceTestTask(name, clazz, performanceSourceSet, prepareSamplesTask).configure(configure)
         }
 
+        val channelSuffix = if (OperatingSystem.current().isLinux) "" else "-windows"
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt:273]: Can we use something like `OperatingSystem.current().name.toLowerCase()` or something so that this works for macOS as well?

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt [hunk 9]
```diff
@@ -57,7 +58,7 @@ abstract class BuildCommitDistribution : DefaultTask() {
     fun buildCommitDistribution() {
         val rootProjectDir = project.rootProject.rootDir.absolutePath
         val commit = commitBaseline.map { it.substring(it.lastIndexOf('-') + 1) }
-        tryBuildDistribution(RemoteProject.checkout(this, rootProjectDir, commit.get()))
+        tryBuildDistribution(RemoteProject.checkout(this, rootProjectDir, commit.get(), project.rootProject.buildDir.resolve("bcd")))
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt:61]: Can we use a longer, more descriptive name? Like `commit-dist`?

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt [hunk 10]
```diff
@@ -50,7 +51,7 @@ abstract class DetermineBaselines @Inject constructor(@get:Internal val distribu
             determinedBaselines.set(defaultBaseline)
         } else if (configuredBaselines.getOrElse("") == flakinessDetectionCommitBaseline) {
             determinedBaselines.set(determineFlakinessDetectionBaseline())
-        } else if (!currentBranchIsMasterOrRelease() && configuredBaselines.isDefaultValue()) {
+        } else if (!currentBranchIsMasterOrRelease() && !OperatingSystem.current().isWindows && configuredBaselines.isDefaultValue()) {
             determinedBaselines.set(forkPointCommitBaseline())
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt:55]: Why not do this on Windows, and what's the consequence of this?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/buildcache/TaskOutputCachingNativePerformanceTest.groovy [hunk 11]
```diff
@@ -17,13 +17,16 @@
 package org.gradle.performance.regression.buildcache
 
 import org.gradle.initialization.ParallelismBuildOptions
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import spock.lang.Unroll
 
+@Requires(TestPrecondition.NOT_WINDOWS)
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/buildcache/TaskOutputCachingNativePerformanceTest.groovy:24]: I think this should require Linux specifically instead.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/kotlindsl/KotlinDslVsGroovyDslPerformanceTest.groovy [hunk 12]
```diff
@@ -35,8 +35,8 @@ class KotlinDslVsGroovyDslPerformanceTest extends AbstractCrossBuildPerformanceT
 
         given:
         runner.testGroup = 'Kotlin DSL vs Groovy DSL'
-        def groovyDslBuildName = 'Groovy DSL build'
-        def kotlinDslBuildName = 'Kotlin DSL build'
+        def groovyDslBuildName = 'Groovy_DSL_build'
+        def kotlinDslBuildName = 'Kotlin_DSL_build'
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/kotlindsl/KotlinDslVsGroovyDslPerformanceTest.groovy:39]: What was wrong with the spaces here?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftCleanBuildPerformanceTest.groovy [hunk 13]
```diff
@@ -19,15 +19,18 @@ package org.gradle.performance.regression.nativeplatform
 import org.gradle.initialization.ParallelismBuildOptions
 import org.gradle.performance.AbstractCrossVersionGradleProfilerPerformanceTest
 import org.gradle.performance.categories.SlowPerformanceRegressionTest
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import org.junit.experimental.categories.Category
 import spock.lang.Unroll
 
 @Category(SlowPerformanceRegressionTest)
+@Requires(TestPrecondition.NOT_WINDOWS)
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftCleanBuildPerformanceTest.groovy:28]: I think it would be better to say that this is supposed to work on Linux and macOS.

**Review by lptr — CHANGES_REQUESTED** [review]
Thanks @blindpirate, this is getting along nicely. I left a few comments.

**Comment by lptr** [comment]
Sorry, not sure how I managed to close this. :/

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/kotlindsl/KotlinDslVsGroovyDslPerformanceTest.groovy (hunk 12 above)
Comment (blindpirate) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/kotlindsl/KotlinDslVsGroovyDslPerformanceTest.groovy:39]: There're issues when these spaces are passed as system properties, something like `GRADLE_OPTS=-DgradleUserHome=Groovy DSL Build`

so I simply removed the space.

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt (hunk 10 above)
Comment (blindpirate) [reply buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt]: I don't know, let's remove it to see if anything breaks. My assumption is that it ran into a long path issue.

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt (hunk 9 above)
Comment (blindpirate) [reply buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt]: I don't know what's `bcd`, I removed it.

File: .teamcity/common/Os.kt (hunk 6 above)
Comment (blindpirate) [reply .teamcity/common/Os.kt]: Done.

File: .teamcity/Gradle_Util_Performance/buildTypes/AdHocPerformanceScenarioWindows.kt (hunk 5 above)
Comment (blindpirate) [reply .teamcity/Gradle_Util_Performance/buildTypes/AdHocPerformanceScenarioWindows.kt]: Done on `master`.

File: .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt (hunk 4 above)
Comment (blindpirate) [reply .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt]: Done.

File: .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt (hunk 3 above)
Comment (blindpirate) [reply .teamcity/Gradle_Check/configurations/IndividualPerformanceScenarioWorkers.kt]: done.

File: .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt (hunk 2 above)
Comment (blindpirate) [reply .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt]: done.

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt (hunk 10 above)
Comment (blindpirate) [reply buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt]: Hmmmmm...`git` can't checkout the source because of long path issues...

**Comment by blindpirate** [comment]
@bot-gradle test this

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy [hunk 14]
```diff
@@ -19,13 +19,16 @@ package org.gradle.performance.regression.nativeplatform
 import org.gradle.initialization.ParallelismBuildOptions
 import org.gradle.performance.AbstractCrossVersionGradleInternalPerformanceTest
 import org.gradle.performance.mutator.AbstractFileChangeMutator
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import spock.lang.Unroll
 
+@Requires(TestPrecondition.LINUX)
 class SwiftBuildPerformanceTest extends AbstractCrossVersionGradleInternalPerformanceTest {
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy:27]: Why not run this on macOS too?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftCleanBuildPerformanceTest.groovy [hunk 15]
```diff
@@ -19,15 +19,18 @@ package org.gradle.performance.regression.nativeplatform
 import org.gradle.initialization.ParallelismBuildOptions
 import org.gradle.performance.AbstractCrossVersionGradleProfilerPerformanceTest
 import org.gradle.performance.categories.SlowPerformanceRegressionTest
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import org.junit.experimental.categories.Category
 import spock.lang.Unroll
 
 @Category(SlowPerformanceRegressionTest)
+@Requires(TestPrecondition.LINUX)
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftCleanBuildPerformanceTest.groovy:28]: Why not run this on macOS too?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/buildcache/TaskOutputCachingSwiftPerformanceTest.groovy [hunk 16]
```diff
@@ -17,13 +17,15 @@
 package org.gradle.performance.regression.buildcache
 
 import org.gradle.initialization.ParallelismBuildOptions
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
 import spock.lang.Unroll
 
+@Requires(TestPrecondition.LINUX)
 class TaskOutputCachingSwiftPerformanceTest extends AbstractTaskOutputCachingPerformanceTest {
```
Comment (lptr) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/buildcache/TaskOutputCachingSwiftPerformanceTest.groovy:25]: Why not run this on macOS too?

File: buildSrc/subprojects/performance-testing/src/test/kotlin/gradlebuild/performance/tasks/DetermineBaselinesTest.kt [hunk 17]
```diff
@@ -83,6 +86,7 @@ class DetermineBaselinesTest {
 
     @Test
     fun `determines fork point commit on feature branch and empty configuration`() {
+        Assume.assumeFalse(OperatingSystem.current().isWindows)
```
Comment (lptr) [inline buildSrc/subprojects/performance-testing/src/test/kotlin/gradlebuild/performance/tasks/DetermineBaselinesTest.kt:89]: Let's leave a comment here about why we need to not run this on Windows.

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt (hunk 10 above)
Comment (lptr) [reply buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt:55]: Please explain the reasons for this in a comment.

**Review by lptr — CHANGES_REQUESTED** [review]

File: .teamcity/common/performance-test-extensions.kt (hunk 1 above)
Comment (blindpirate) [reply .teamcity/common/performance-test-extensions.kt]: This is done.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy (hunk 14 above)
Comment (blindpirate) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy:27]: All scenarios are pretty expensive (~6h on mac agents), so my understanding is that we're going to run only a small number of scenarios on macos. The exact subset of every OS can be done in a followup https://docs.google.com/document/d/1F_m6XLOWvpLdaGJ7ey6sta3mgC-HymAYe-EttNwhMss/edit

File: buildSrc/subprojects/performance-testing/src/test/kotlin/gradlebuild/performance/tasks/DetermineBaselinesTest.kt (hunk 17 above)
Comment (blindpirate) [reply buildSrc/subprojects/performance-testing/src/test/kotlin/gradlebuild/performance/tasks/DetermineBaselinesTest.kt:91]: Done.

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt (hunk 9 above)
Comment (lptr) [reply buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/BuildCommitDistribution.kt]: I'm guessing it's from `buildCommitDistribution`.

**Review by lptr — APPROVED** [review]
LGTM!

**Comment by lptr** [comment]
@wolfs PTAL post merge.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy (hunk 14 above)
Comment (lptr) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy:27]: Let's add some documentation to these exclusions. Maybe we should add an optional `because` parameter to `@Requires`. Or we could just leave a comment here.

I think it would make sense to also add explanations to all the other `@Requires` annotations in this PR.

**Comment by wolfs** [comment]
@blindpirate Can you point me to what happened to the original changes on this branch? I can't find the commits any more. It seems you added some of them to `master` already.

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt [hunk 18]
```diff
@@ -269,34 +270,36 @@ class PerformanceTestPlugin : Plugin<Project> {
             createDistributedPerformanceTestTask(name, clazz, performanceSourceSet, prepareSamplesTask).configure(configure)
         }
 
+        val channelSuffix = if (OperatingSystem.current().isLinux) "" else "-${OperatingSystem.current().name.toLowerCase()}"
+
         create("distributedPerformanceTest", DistributedPerformanceTest::class) {
             includeCategories(performanceRegressionTestCategory)
             excludeCategories(slowPerformanceRegressionTestCategory)
-            channel = "commits"
+            channel = "commits$channelSuffix"
```
Comment (wolfs) [inline buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt:278]: Did you update the performance database contents? If you don't, then we'll lose all the history, right?

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt [hunk 19]
```diff
@@ -269,34 +270,36 @@ class PerformanceTestPlugin : Plugin<Project> {
...
             retryFailedScenarios()
         }
         create("distributedSlowPerformanceTest", DistributedPerformanceTest::class) {
             includeCategories(slowPerformanceRegressionTestCategory)
-            channel = "commits"
+            channel = "commits$channelSuffix"
```
Comment (wolfs) [inline buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/PerformanceTestPlugin.kt:283]: Instead of using a postfix for the channel, I think it would be much better to use a separate column in the performance database (e.g. `os`) similar to `branch`. Can you make that change?

File: buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt [hunk 20]
```diff
@@ -24,6 +24,7 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.provider.Property
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
+import org.gradle.internal.os.OperatingSystem
```
Comment (wolfs) [inline buildSrc/subprojects/performance-testing/src/main/kotlin/gradlebuild/performance/tasks/DetermineBaselines.kt:26]: What happened here?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AbstractRealLifeAndroidBuildPerformanceTest.groovy [hunk 21]
```diff
@@ -27,7 +27,7 @@ class AbstractRealLifeAndroidBuildPerformanceTest extends AbstractCrossVersionGr
 
     def setup() {
         runner.args = [AndroidGradlePluginVersions.OVERRIDE_VERSION_CHECK]
-        runner.targetVersions = ["6.7-20200723220251+0000"]
+        runner.targetVersions = ["6.7-20200824220048+0000"]
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/android/AbstractRealLifeAndroidBuildPerformanceTest.groovy:30]: There are a few rebaselined tests in this PR. Why have they been rebaselined? Can you revert the rebaselining on `master`?

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/ParallelDownloadsPerformanceTest.groovy [hunk 22]
```diff
@@ -155,3 +155,4 @@ class ParallelDownloadsPerformanceTest extends AbstractCrossVersionGradleInterna
     }
 
 }
+
```
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/corefeature/ParallelDownloadsPerformanceTest.groovy:158]: Nit: Extra newline in unchanged file.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy (hunk 14 above)
Comment (wolfs) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/nativeplatform/SwiftBuildPerformanceTest.groovy:27]: We really should change how we decide which tests to run on which operating systems. I think adding an `@Requires` is OK for tests which don't run on some OS, but it shouldn't be used to decide what to run on which OS.

For this test, I suppose it should be `@Requires(TestPrecondition.NOT_WINDOWS)`.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/kotlindsl/KotlinDslVsGroovyDslPerformanceTest.groovy (hunk 12 above)
Comment (wolfs) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/kotlindsl/KotlinDslVsGroovyDslPerformanceTest.groovy:39]: How hard is it to fix that we actually pass system properties with spaces correctly on Windows?

**Review by wolfs — COMMENTED** [review]
I left some comments. Shall I create an issue to fix those?
